### ISSUE TITLE
Unable to change cursor type/shape in different modes - vi

### ISSUE THREAD

**Issue by laxisforte** [fish-shell/fish-shell#6478]
`fish --version`
Fish version 3.02

`echo $TERM`
Xterm-256color

`uname -a`
Linux xxxxxx 4.4.0-18362-Microsoft #476-Microsoft Fri Nov 01 16:53:00 PST 2019 x86_64 x86_64 x86_64 GNU/Linux

`sh -c 'env HOME=$(mktemp -d) fish'`
The problem persists

- Windows 10 version 1909
- Windows new terminal version 0.7.345.10

fish_user_key_bindings.fish

```javascript
function fish_user_key_bindings
  fish_vi_key_bindings
  fzf_key_bindings
  set -g fish_cursor_insert line
  set -g fish_cursor_replace_one underscore
  set -g fish_cursor_visual block
  bind -M insert -m default jj backward-char force-repaint
  bind -M insert hh forward-word force-repaint
  bind -M insert \r 'replace_then_execute'
end
```

config.fish

```javascript
set -U fish_greeting ""
set theme_color_scheme dracula
set -g FZF_CTRL_T_COMMAND "fd --type f --hidden --follow --exclude .git"
```
Cursor shape (solid box) is same in all modes...
Any suggestion ?
Thank you,
P.

**Comment by faho** [comment]
We only change the cursor if we think the terminal can do it. I've not looked at the new windows terminal. Can it change the cursor? How can we detect it? (Note that I'm pretty sure that like 95% of all terminals it sets $TERM to xterm-256color, so that's out)

**Comment by faho** [comment]
@laxisforte Can you tell us what `set -q WT_SESSION && echo YES $WT_SESSION` prints in the new terminal and ideally the old one?

**Comment by faho** [comment]
Fixed by 76bbcb9804e7e231b6f382bcebd9f8ca4d833b72
